polkapox/1.0.0-beta/Dockerfile

```diff
@@ -0,0 +1,85 @@
+##### Building stage #####
+
+# Stage 1: Build Python environment with required packages
+FROM python:3.8-slim AS app
+
+# Set environment variables
+ENV PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1
+
+# Add metadata
+LABEL authors="CDC PolkaPox Team"
+LABEL description="Container used in the PolkaPox pipeline by graph_reconstruct.nf module"
+LABEL base.image="python:3.8-slim"
+LABEL dockerfile.version="1.0"
+LABEL website="https://github.com/CDCgov/polkapox"
+LABEL license="https://github.com/CDCgov/polkapox/blob/master/LICENSE"
+
+# Install required system dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    build-essential \
+    libopenblas-dev \
+    wget \
+    procps
+
+# Install Python packages
+RUN pip install --no-cache-dir \
+    biopython==1.83 \
+    numpy==1.24.4 \
+    mkl==2024.0.0 \
+    networkx==3.1 \
+    gfapy==1.2.3
+
+# Stage 2: Build the final (lightweight) image 
```

Comment:
I think it's a great idea to try and save space, but I don't think you're saving a lot of space in this situation. I'd merge app and app2 into the same stage.